processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -39,14 +48,29 @@ public SegmentMetadataQuery(
       @JsonProperty("intervals") QuerySegmentSpec querySegmentSpec,
       @JsonProperty("toInclude") ColumnIncluderator toInclude,
       @JsonProperty("merge") Boolean merge,
-      @JsonProperty("context") Map<String, Object> context
+      @JsonProperty("context") Map<String, Object> context,
+      @JsonProperty("usingDefaultInterval") Boolean useDefaultInterval
   )
   {
-    super(dataSource, querySegmentSpec, context);
+    super(
+        dataSource,
+        (querySegmentSpec == null) ? new MultipleIntervalSegmentSpec(Arrays.asList(DEFAULT_INTERVAL))
+                                   : querySegmentSpec,
+        context
+    );
+
+    if (querySegmentSpec == null) {
+      this.usingDefaultInterval = true;
+    } else {
+      this.usingDefaultInterval = useDefaultInterval == null ? false : useDefaultInterval;
+    }
```

Comment:
IMO we should not document usingDefaultInterval, its only purpose is to help "remember" that a user did not specify intervals. Without something like that it's hard to remember, since the querySegmentSpec needs to be filled in with DEFAULT_INTERVAL to allow the later filtering to work.

So I think if a user specifies "usingDefaultInterval" directly that is their own problem and we don't need to provide any particular behavior :)